contrib/oadp-recovery/manifests/cronjob.yaml

```diff
@@ -0,0 +1,412 @@
+        if ! last_backup=$(find_last_related_backup "$cluster_name" "$cluster_namespace"); then
+            log_verbose "No related backups found for OADP-paused cluster $cluster_name, should unpause"
+            return 0
+        fi
+
+        log_debug "Found last related backup: $last_backup"
+
+        # Check if the backup is in terminal state
+        local backup_phase
+        if backup_phase=$(is_backup_in_terminal_state "$last_backup" "$OADP_NAMESPACE"); then
+            log_verbose "Last backup $last_backup is in terminal state ($backup_phase) - should unpause cluster $cluster_name"
+            return 0
+        fi
+
+        log_verbose "Last backup $last_backup is still in progress - keeping cluster $cluster_name paused"
+        return 1
+    }
+
+    # Function to resume cluster from OADP pause
+    resume_cluster_from_oadp() {
+        local cluster_name="$1"
+        local cluster_namespace="$2"
+
+        log_info "Resuming cluster $cluster_name from OADP pause"
+
+        # Remove OADP annotations from HostedCluster
+        if [[ "$DRY_RUN" == "true" ]]; then
+            log_info "DRY RUN: Would remove OADP annotations and unpause HostedCluster $cluster_name"
+        else
+            kubectl annotate hostedcluster "$cluster_name" -n "$cluster_namespace" \
+                "${OADP_PAUSED_BY_ANNOTATION}-" "${OADP_PAUSED_AT_ANNOTATION}-" || {
+                log_error "Failed to remove OADP annotations from HostedCluster $cluster_name"
+                return 1
+            }
+
+            # Clear pausedUntil field
+            kubectl patch hostedcluster "$cluster_name" -n "$cluster_namespace" \
+                --type='merge' -p='{"spec":{"pausedUntil":null}}' || {
+                log_error "Failed to clear pausedUntil field for HostedCluster $cluster_name"
+                return 1
+            }
+
+            log_info "Successfully resumed HostedCluster $cluster_name"
+        fi
+
+        # Get NodePools for this cluster
+        local nodepools
+        if nodepools=$(kubectl get nodepools -n "$cluster_namespace" \
+            -o jsonpath="{.items[?(@.spec.clusterName=='$cluster_name')].metadata.name}" 2>/dev/null); then
+
+            for nodepool in $nodepools; do
+                if [[ -z "$nodepool" ]]; then
+                    continue
+                fi
+
+                log_info "Resuming NodePool $nodepool for cluster $cluster_name"
+
+                if [[ "$DRY_RUN" == "true" ]]; then
+                    log_info "DRY RUN: Would remove OADP annotations and unpause NodePool $nodepool"
+                else
+                    # Remove OADP annotations from NodePool
+                    kubectl annotate nodepool "$nodepool" -n "$cluster_namespace" \
+                        "${OADP_PAUSED_BY_ANNOTATION}-" "${OADP_PAUSED_AT_ANNOTATION}-" 2>/dev/null || true
+
+                    # Clear pausedUntil field
+                    kubectl patch nodepool "$nodepool" -n "$cluster_namespace" \
```

Comment:
Should this just continue with the next NodePool in the loop, instead of failing immediately?

Reply:
Yes, if that is not being unpaused it will the error log and continue with the next NP, if we fail, that could block the unpause of multiple NP in the HC.

I prefer one fails and could be unpaused in the next try than just block it and keep paused the rest of the clusters :/

Reply:
I see this in the script:
```
kubectl patch nodepool "$nodepool" -n "$cluster_namespace" \
                    --type='merge' -p='{"spec":{"pausedUntil":null}}' || {
                    log_error "Failed to clear pausedUntil field for NodePool $nodepool"
                    return 1
                }
```
And it calls `return 1` which will break the loop and will NOT continue with next NodePool. Do we have the same understanding?